packages/components/inputs/money-input/src/money-input.tsx

```diff
@@ -319,17 +360,149 @@ const formatAmount = (rawAmount, currencyCode, locale) => {
+const getAmountInputName = (name?: string) =>
+  name ? `${name}.amount` : undefined;
+const getCurrencyDropdownName = (name?: string) =>
   name ? `${name}.currencyCode` : undefined;
 
-const MoneyInput = (props) => {
+type TValue = {
+  amount: string;
+  currencyCode: TCurrencyCode;
+};
+
+type TEvent = {
+  target: {
+    id?: string;
+    name?: string;
+    value?: string | string[] | null;
+  };
+  persist?: () => void;
+};
+
+type TMoneyInputProps = {
+  /**
+   * Used as HTML id property. An id is auto-generated when it is not specified.
+   */
+  id?: string;
+  /**
+   * Used as HTML `autocomplete` property
+   */
+  autoComplete?: string;
+  /**
+   * The prefix used to create a HTML `name` property for the amount input field (`${name}.amount`) and the currency dropdown (`${name}.currencyCode`).
+   */
+  name?: string;
+  /**
+   * Value of the input. Consists of the currency code and an amount. `amount` is a string representing the amount. A dot has to be used as the decimal separator.
+   */
+  value: TValue;
+  /**
+   * List of possible currencies. When not provided or empty, the component renders a label with the value's currency instead of a dropdown.
+   */
+  currencies: string[];
+  /**
+   * Placeholder text for the input
+   */
+  placeholder?: string;
+  /**
+   * Called when input is blurred
+   */
+  onBlur?: (event: TEvent) => void;
+  /**
+   * Called when input is focused
+   */
+  onFocus?: (event: TEvent) => void;
+  /**
+   * Indicates that the input cannot be modified (e.g not authorized, or changes currently saving).
+   */
+  isDisabled?: boolean;
+  /**
+   * Indicates that the field is displaying read-only content
+   */
+  isReadOnly?: boolean;
+  /**
+   * Focus the input on initial render
+   */
+  isAutofocussed?: boolean;
+  /**
+   * Called with the event of the input or dropdown when either the currency or the amount have changed.
+   * <br />
+   * Signature: `(event) => void`
+   */
+  onChange: (event: TEvent) => void;
+  /**
+   * Dom element to portal the currency select menu to
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  menuPortalTarget?: ReactSelectProps['menuPortalTarget'];
+  /**
+   * z-index value for the currency select menu portal
+   */
+  menuPortalZIndex?: number;
+  /**
+   * whether the menu should block scroll while open
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  menuShouldBlockScroll: ReactSelectProps['menuShouldBlockScroll'];
+  /**
+   * Indicates that input has errors
+   */
+  hasError?: boolean;
+  /**
+   * Control to indicate on the input if there are selected values that are potentially invalid
+   */
+  hasWarning?: boolean;
+  /**
+   * Shows high precision badge in case current value uses high precision.
+   */
+  hasHighPrecisionBadge?: boolean;
+  /**
+   * Horizontal size limit of the input fields.
+   */
+  horizontalConstraint?:
```

Comment:
Maybe the available values should start in 1?
I'm saying because of what I read [here](https://github.com/commercetools/ui-kit/pull/2062#discussion_r783883512).

Reply:
Oh.. No. It doesn't have to start from 1. 
It just needs to be consisted with the previous proptTypes. 
Btw, I actually made a mistake with the comment [here](https://github.com/commercetools/ui-kit/pull/2062#discussion_r783883512).